tfhe/src/high_level_api/shortints/server_key.rs

```diff
@@ -429,7 +429,9 @@ where
     where
         F: Fn(u64) -> u64,
     {
-        let lookup_table = self.key.generate_lookup_table(func);
+        let lookup_table = self
+            .key
+            .generate_lookup_table(|a| func(a) % self.key.message_modulus.0 as u64);
```

Comment:
should this be the message_modulus or the max modulus including the carry ? as long as we are not trashing the padding bit ? same bellow

Reply:
I think it might be better to have it include the carry space also since using the carry space is inherent to the proper use of shortint :thinking:

Reply:
I guess it's moslty an interpretation of what users should expect from this function

But I agree, I could make more sense to include the carry modulus

Reply:
I guess an unchecked_generated_lut would be nice ? and maybe a smart_generate_lut which would give the msg and carry values basically would do

```rust
for plaintext in 0..plaintext_mod {
    let msg = plaintext % msg_mod;
    let carry = plaintext / msg_mod;
    let lut_val = user_provided_fn(msg, carry);
    // store in lut
}
```

this way you can chose if you are manipulating the message and/or the carry

Reply:
Shortints hve been removed for the HL API so I guess this solves this issue